docs/resources/resource_policy.md

```diff
@@ -0,0 +1,145 @@
+
+-> **NOTE**: Resource Policies are currently in Public Preview. You will need to
+1. Enable the `Atlas Resource Policies` Beta Feature in your organization (contact MongoDB Support).
+2. Enable the [Preview Features](https://github.com/mongodb/terraform-provider-mongodbatlas?tab=readme-ov-file#preview-features) when running `terraform` commands.
+
+
+## Example Usages
+
+```terraform
+resource "mongodbatlas_resource_policy" "project_ip_access_list" {
+  org_id = var.org_id
+  name   = "forbid-access-from-anywhere"
+
+  policies = [
+    {
+      body = <<EOF
+        forbid (
+                principal,
+                action == cloud::Action::"project.edit",
+                resource
+        )
+                when {
+                context.project.ipAccessList.contains(ip("0.0.0.0/0"))
+        };
+EOF
+    },
+  ]
+}
+
+resource "mongodbatlas_resource_policy" "cloud_provider" {
+  org_id = var.org_id
+  name   = "forbid-cloud-provider"
+  policies = [
+    {
+      body = templatefile("${path.module}/cloud-provider.cedar", {
+        CLOUD_PROVIDER = "azure"
+      })
+    },
+    {
+      body = templatefile("${path.module}/cloud-provider.cedar", {
+        CLOUD_PROVIDER = "aws"
+      })
+    },
+  ]
+}
+
+data "cedar_policyset" "cloud_region" {
+  policy {
+    any_principal = true
+    effect        = "forbid"
+    action = {
+      type = " cloud::Action"
+      id   = "cluster.createEdit"
+    }
+    any_resource = true
+    when {
+      text = "context.cluster.regions.contains(cloud::region::\"gcp:us-east1\")"
+    }
+  }
+}
+
+resource "mongodbatlas_resource_policy" "cloud_region" {
+  org_id = var.org_id
+  name   = "forbid-cloud-region"
+  policies = [
+    {
+      body = data.cedar_policyset.cloud_region.text
+    },
+  ]
+}
+
+
+data "mongodbatlas_resource_policy" "project_ip_access_list" {
+  org_id = mongodbatlas_resource_policy.project_ip_access_list.org_id
+  id     = mongodbatlas_resource_policy.project_ip_access_list.id
+}
+
+data "mongodbatlas_resource_policies" "this" {
+  org_id = data.mongodbatlas_resource_policy.project_ip_access_list.org_id
+
+  depends_on = [mongodbatlas_resource_policy.project_ip_access_list, mongodbatlas_resource_policy.cloud_provider, mongodbatlas_resource_policy.cloud_region]
+}
+
+
+output "policy_ids" {
+  value = { for policy in data.mongodbatlas_resource_policies.this.resource_policies : policy.name => policy.id }
+}
+```
+
+<!-- schema generated by tfplugindocs -->
+## Schema
+
+### Required
+
+- `name` (String) Human-readable label that describes the atlas resource policy.
+- `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.
+- `policies` (Attributes List) List of policies that make up the atlas resource policy. (see [below for nested schema](#nestedatt--policies))
```

Comment:
```suggestion
- `name` (String) Human-readable label that describes the Atlas resource policy.
- `org_id` (String) Unique 24-hexadecimal digit string that identifies the organization that contains your projects. Use the [/orgs](#tag/Organizations/operation/listOrganizations) endpoint to retrieve all organizations to which the authenticated user has access.
- `policies` (Attributes List) List of policies that make up the Atlas resource policy. (see [below for nested schema](#nestedatt--policies))
```

Reply:
Changed in a8851acdb8c44933b67c874a3160f5b6938a5ba2